### PR TITLE
Remove jsts from turf/union

### PR DESCRIPTION
**Pull request by rowanwins**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Have read [How To Contribute](https://github.com/Turfjs/turf/blob/master/CONTRIBUTING.md#how-to-contribute).
- [x] Run `npm test` at the sub modules where changes have occurred.
- [x] Run `npm run lint` to ensure code style at the turf module level.

Have made a start on this but am less certain about how to proceed. So for a basic union with only 2 polys things work fine. But with jsts we were able to pass in a featurecollection of lots of polys and so test cases 2 & 3 are currently failing. 

Attempting to convert a feature collection into a multipoly and then pass that into martinez is also not producing the desired result.

So more digging required but I thought I'd dump the initial work here so ppl know something is happening 

cc @w8r

### CODE REVIEW

**Comment by DenisCarriere** [comment]
@rowanwins I'm 100% happy with your proposed two input `polygon1` & `polygon2` implementation, this gives us room for having `options` as a 3rd parameter.

👍 Going to look at this PR further, so far so good!

File: packages/turf-union/index.js [hunk 1]
```diff
@@ -27,22 +30,15 @@ import { GeoJSONReader, GeoJSONWriter, UnionOp } from 'turf-jsts';
  * //addToMap
  * var addToMap = [poly1, poly2, union];
  */
-function union() {
-    var reader = new GeoJSONReader();
-    var result = reader.read(JSON.stringify(arguments[0].geometry));
+function union(polygon1, polygon2) {
+    var geom1 = getGeom(polygon1);
+    var geom2 = getGeom(polygon2);
+    var properties = polygon1.properties || {};
```
Comment (DenisCarriere) [inline packages/turf-union/index.js]: 🤔 Tricky one... We should allow the user to override these properties from the `options` param.

File: packages/turf-union/index.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
-import { GeoJSONReader, GeoJSONWriter, UnionOp } from 'turf-jsts';
+import martinez from 'martinez-polygon-clipping';
+import { getGeom } from '@turf/invariant';
+import { multiPolygon, polygon } from '@turf/helpers';
 
 /**
  * Takes two or more {@link Polygon|polygons} and returns a combined polygon. If the input polygons are not contiguous, this function returns a {@link MultiPolygon} feature.
```
Comment (DenisCarriere) [inline packages/turf-union/index.js]: Note: I've added more links to the Documentation YAML config.

- `{@link Polygon|polygons}` => `{@link (Multi)Polygon(s)}`

https://github.com/Turfjs/turf/blob/master/documentation.yml#L177

**Comment by rowanwins** [comment]
I have a pull request open in Martinez to support mutipoly in the inputs. This still needs a bit more investigation but hopefully should be sorted in the next week.

File: packages/turf-union/package.json [hunk 3]
```diff
@@ -42,7 +42,10 @@
     "write-json-file": "*"
   },
   "dependencies": {
+    "@turf/combine": "^5.1.5",
     "@turf/helpers": "^5.1.5",
+    "@turf/invariant": "^6.0.0-beta.3",
```
Comment (DenisCarriere) [inline packages/turf-union/package.json]: @rowanwins I guess this release "breaks" stuff... 😠 

```
lerna ERR! Error [ERR_REQUIRE_ESM]: Must use @std/esm to load ES Module: /home/travis/build/Turfjs/turf/packages/turf-union/node_modules/@turf/invariant/index.mjs
```

I'll revert those `beta` releases

**Pull request merged**